### ISSUE TITLE
Switching Audio output during gaming session can potentially render the games frozen

### ISSUE THREAD

**Issue by RebelliousX** [LizardByte/Sunshine#3430]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Is your issue described in the documentation?

- [x] I have read the documentation

### Is your issue present in the latest beta/pre-release?

None

### Describe the Bug

It happens with some games for example Prince of Persia Forgotten Sands from Steam.
The game runs fine, but if I exit the session while the game is still running, trying to join the session again
the game is frozen, but streaming still happening. And if I exit the session in the game menu, and resume
the session, the animation of the main menu of the game is frozen, but the mouse still active.

In task manager, it shows the process of Prince of Persia is not responding with 0 percent of activity.

This started happening about a month or so ago. I believe around the commit where resuming session or some audio changes
I really can't remember exactly. But less than 2 months for sure. I wish I have access to prior `pre-release builds`. The actions / runs doesn't have any installers or builds to test. I can't compile the code unfortunately.

I thought this is due to recent Windows 11 24H2, so I wiped the system and installed latest Windows 10 22H2, same issue.

Nvidia driver v566.14, I downgraded to a 2 month older driver v561.09. I tried both Game Ready and Studio drivers, same thing.

### Expected Behavior

To be able to resume the session without crashing the game

### Additional Context

I tried Intel encoder, software encoder. Same thing.

I tried installing Virtual Display Driver, the game didn't freeze, but since I was selecting the Virtual monitor and the game starts on the default monitor, I couldn't see anything except the wallpaper of monitor 2. But the game didn't freeze.

### Host Operating System

Windows

### Operating System Version

Tried both Windows 11 v24H2 and Windows 10 v22H2 fully updated

### Architecture

amd64/x86_64

### Sunshine commit or version

v2024.1204.170534

### Package

Windows - installer (recommended)

### GPU Type

none (software encoding), hardware encoding, nothing seems to work.

### GPU Model

GTX 1050ti / Intel 630

### GPU Driver/Mesa Version

see above

### Capture Method

I tried both DXGI (running as service) and WGC (running as admin, not as service), still same issue.

### Relevant log output

```shell
output doesn't show any errors.
```

**Comment by RebelliousX** [comment]
I installed Geforce Experience and enabled Game Stream, there are no freezes for the games. With sunshine, the games freezes whether it is a windowed or full-screen mode (more games have this now), but only when resuming the session. Geforce Experience has low frame-rates in 30s compared to sunshine solid 60 for some reason.

I hope this gets fixed, I tried everything, disabling HAGS, realtime priority. I am not even using GTX 1050ti to encode, I used Intel iGPU and even Software Encoding, same thing.

**Comment by RebelliousX** [comment]
Update, I was able to pinpoint the cause of this. It seems lots of games don't like it when audio output device change while the game is running.

A workaround for now is to have Moonlight to `play audio on host`, which prevents the audio switching during game play. The problem is I don't need audio on hos (main PC). So I downloaded `SetVol` app and added command for `DO` and `Undo` like

For Do:
```
C:\temp\SetVol mute
```

For Undo:
```
 C:\temp\SetVol unmute
```

Which works great. The issue is, the UNDO command doesn't run when the `client gets disconnected`, but only when `session quits`. That is, if I minimize Moonlight, the client disconnects, streaming stops, but Volume still muted. If I open Moonlight and force session to quit, the `Undo` command runs. 

I also have similar issue when running VDD trying to run UNDO command when client disconnect, for example Moonlight on Shield TV (session is on, just Moonlight minimized). The Undo command to switch back to main display doesn't run until I force quit the session from Moonlight.

**Comment by lukasstraub2** [comment]
TLDR: Workaround is to enable `play audio on host` on the moonlight client.
